library/std/src/personality.rs

```diff
@@ -0,0 +1,46 @@
+//! This module contains the implementation of the `eh_personality` lang item.
+//!
+//! The actual implementation is heavily dependent on the target since Rust
+//! tries to use the native stack unwinding mechanism whenever possible.
+//!
+//! This personality function is still required with `-C panic=abort` because
+//! it is used to catch foreign exceptions from `extern "C-unwind"` and turn
+//! them into aborts.
+//!
+//! Additionally, ARM EHABI uses the personality function when generating
+//! backtraces.
+
+mod dwarf;
+
+#[cfg(not(test))]
+cfg_if::cfg_if! {
+    if #[cfg(target_os = "emscripten")] {
+        mod emcc;
+    } else if #[cfg(target_env = "msvc")] {
+        // This is required by the compiler to exist (e.g., it's a lang item),
+        // but it's never actually called by the compiler because
+        // _CxxFrameHandler3 is the personality function that is always used.
```

Comment:
It looks like this comment isn't directly replicated from the previous (which was a little less specific in terms of which cxx function is used) - some cursory online searching suggests that we may actually be using the 4th version these days, not 3rd?

Reply:
The old comment was incorrect, we now always use `_CxxFrameHandler3` for our personality function on MSVC targets.

`_CxxFrameHandler4` exists but it is not supported by LLVM, only by MSVC itself.